src/main/java/com/atlassian/db/replica/spi/Chief.java

```diff
@@ -0,0 +1,14 @@
+package com.atlassian.db.replica.spi;
+
+import com.atlassian.db.replica.internal.RouteDecisionBuilder;
+
+import java.sql.Connection;
+import java.util.Collection;
+import java.util.function.Supplier;
+
+/**
+ * Can override the automatic decision to make a final call where the query should go.
+ */
+public interface Chief {
```

Comment:
```suggestion
/**
 * Can reroute the current call by overriding the current decision.
 */
public interface Router {
```

Impls would go `SqlQueryChief` -> `SqlQueryRouter`, `ConsistencyGuardingChief` -> `ConsistencyRouter`.